### PR TITLE
VAOS: Migrate MFS Clinics Search Call Over to VAOS Service

### PR DESCRIPTION
**Pull request by ajmagdub**
## Summary
We have updated the system for obtaining clinic information from the Mobile Facility Service (MFS) to the VAOS Service. Update the VAOS::V2::MobileFacilityService#get_clinic method to use the "/vaos/v1/locations/<site_id>/clinics" endpoint, which points to the VAOS Service.

Additionally, the assignments for the friendly name in VAOS::V2::AppointmentsController were modified to retrieve
its value from the clinics service name. The VAOS service provides the friendly name in the service name, if
available, otherwise the internal clinic will be returned. 

## Related issue(s)
department-of-veterans-affairs/va.gov-team#58434

## Testing done
I have made adjustments to the RSpec tests and VCR Cassettes to accommodate the recent change.

### CODE REVIEW

**Review by cferris32 — APPROVED** [review]
LGTM

File: modules/vaos/app/services/vaos/v2/mobile_facility_service.rb [hunk 1]
```diff
@@ -14,12 +14,12 @@ class MobileFacilityService < VAOS::SessionService
       # @return [OpenStruct] An OpenStruct object containing information about the clinic.
       #
       def get_clinic(station_id:, clinic_id:)
-        params = {}
+        params = { clinicIds: clinic_id }
```
Comment (kpethtel) [inline modules/vaos/app/services/vaos/v2/mobile_facility_service.rb:17]: So is it possible to send multiple ids? That could actually be useful to the mobile team.

**Review by kpethtel — APPROVED** [review]
The change looks fine. Per my comment below, would it make sense to add a get_clinics method? Maybe even make a "get_clinics_with_cache" method so that we can get cache optimizations. 

Speaking of which, I wasn't aware of the `get_clinic_with_cache` method. We aren't using this. So thanks for pointing this out. I'll talk to Jayson about changing our code to use it.

File: modules/vaos/app/services/vaos/v2/mobile_facility_service.rb (hunk 1 above)
Comment (kpethtel) [reply modules/vaos/app/services/vaos/v2/mobile_facility_service.rb:17]: Thanks, AJ! Just let us know. I didn't realize your team only recently added those "with cache" methods and that Corey updated our proxy to use the cache version of the get facility method. Thanks for doing that. I'll try to update us to use the clinic cache as well. That may largely make the "get clinics in bulk" method irrelevant, but it seems like a reasonable thing to have.

**Review by jeremy6d — APPROVED** [review]

**Pull request merged**